hadoop-yarn-project/hadoop-yarn/hadoop-yarn-api/src/main/java/org/apache/hadoop/yarn/conf/YarnConfiguration.java

```diff
@@ -4367,7 +4367,7 @@ public static boolean isAclEnabled(Configuration conf) {
    * the default value is 0s.
    */
   public static final long DEFAULT_ROUTER_USER_CLIENT_THREAD_POOL_KEEP_ALIVE_TIME =
-      TimeUnit.SECONDS.toMillis(0); // 0s
+      TimeUnit.SECONDS.toMillis(30); // 0s
```

Comment:
The comment needs to be updated.
Also not sure what's the agreement on changing default values.

Reply:
Thank you for discovering this issue! It was my oversight, and I introduced this problem while conducting integration testing. I will fix it.